File: src/main/java/org/myorg/model/MySource.java
package org.myorg.model;

import org.apache.flink.streaming.api.functions.source.SourceFunction;

import java.util.ArrayList;
import java.util.Random;

public class MySource implements SourceFunction<String> {

    private long count = 1L;
    private boolean isRunning = true;

    @Override
    public void run(SourceContext<String> sourceContext) throws Exception {
        while (isRunning) {
            ArrayList<String> stringList = new ArrayList<>();

            stringList.add("world");
            stringList.add("Flink");
            stringList.add("Stream");
            stringList.add("Batch");
            stringList.add("Table");
            stringList.add("SQL");
            stringList.add("Hello");

            int size = stringList.size();

            int i = new Random().nextInt(size);
            sourceContext.collect(stringList.get(i));

            // 每秒产生一条数据
            Thread.sleep(1000);
        }
    }

    @Override
    public void cancel() {
        isRunning = false;
    }
}

File: src/main/scala/com/it/model/MySource.scala
package com.it.model

import org.apache.flink.streaming.api.functions.source.{RichSourceFunction, SourceFunction}

import scala.collection.mutable.ListBuffer
import scala.util.Random

/**
 * 自定义数据源
 */
class MySource extends RichSourceFunction[String] {

  var count = 1L
  var isRunning = true

   override def run(sourceContext: SourceFunction.SourceContext[String]): Unit = {

    while (isRunning) {
      //      val list: List[String] = List[String]()

      val buffer = new ListBuffer[String]()

      buffer.append("world")
      buffer.append("Flink")
      buffer.append("Table")
      buffer.append("SQL")
      buffer.append("hello")

      val i = Random.nextInt(buffer.size)
      sourceContext.collect(buffer.apply(i))

      Thread.sleep(1000)
    }

  }

  override def cancel(): Unit = {
    isRunning = false
  }

}
